wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/DeferredCommand.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj2.command;
+
+import static edu.wpi.first.util.ErrorMessages.requireNonNullParam;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * Runs the given command when this command is initialized, and ends when it ends. Useful for
+ * performing runtime tasks before creating a new command. If this command is interrupted, it will
+ * cancel the command.
+ *
+ * <p>This class is provided by the NewCommands VendorDep
+ */
+public class DeferredCommand extends Command {
+  private final Supplier<Command> m_supplier;
+  private Command m_command;
+
+  /**
+   * Creates a new DeferredCommand that runs the supplied command when initialized, and ends when it
+   * ends. Useful for lazily creating commands at runtime. The {@link Supplier} will be called each
+   * time this command is initialized. The Supplier <i>must</i> create a new Command each call.
+   *
+   * @param supplier The command supplier
+   * @param requirements The command requirements. This is a {@link Set} to prevent accidental
+   *     omission of command requirements.
+   */
+  public DeferredCommand(Supplier<Command> supplier, Set<Subsystem> requirements) {
+    m_supplier = requireNonNullParam(supplier, "supplier", "DeferredCommand");
+    addRequirements(requirements.toArray(new Subsystem[0]));
+    resetInternalCommand();
+  }
+
+  @Override
+  public void initialize() {
+    var cmd = m_supplier.get();
+    if (cmd != null) {
+      m_command = cmd;
+    }
+    CommandScheduler.getInstance().registerComposedCommands(m_command);
```

Comment:
I'm not sure `registerComposedCommands` checks if the command is already composed.
Also, I'm a bit apprehensive about throwing in cases of double compositions -- but I'm not sure what else can be done.

Reply:
It does check. (See the docs at [CommandScheduler.registerComposedCommands()](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/CommandScheduler.html#registerComposedCommands(edu.wpi.first.wpilibj2.command.Command...)), and the check is at [CommandScheduler.java#L552](https://github.com/wpilibsuite/allwpilib/blob/main/wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java#L552), until the lines shift with the next PR)

Reply:
> Also, I'm a bit apprehensive about throwing in cases of double compositions -- but I'm not sure what else can be done.
Yeah, I am too. I don't love it but the alternative is risking weird hard to debug issues.